core/interop/src/modules/json.rs

```diff
@@ -0,0 +1,67 @@
+#![cfg(feature = "json")]
+//! Functions to create [`Module`]s that expose JSON data directly
+//! as the default export.
+#![allow(clippy::module_name_repetitions)]
+
+use boa_engine::{js_string, Context, JsValue, Module};
+
+/// Create a module that exports a single JSON value as the default export, from its
+/// JSON string. This required the `json` feature to be enabled.
+///
+/// # Errors
+/// This will return an error if the JSON string is invalid or cannot be converted.
+pub fn json_string_module(json: &str, context: &mut Context) -> boa_engine::JsResult<Module> {
+    let json_value = serde_json::from_str::<serde_json::Value>(json).map_err(|e| {
+        boa_engine::JsError::from_opaque(js_string!(format!("Failed to parse JSON: {}", e)).into())
+    })?;
+    let value: JsValue = JsValue::from_json(&json_value, context)?;
+    Ok(Module::from_value_as_default(value, context))
+}
```

Comment:
I'd say just move this to the main crate, since we need to implement `Module::json` anyways.

Reply:
Done. Had to rework the test but that's fine.